rockcraft/pebble.py

```diff
@@ -18,11 +18,142 @@
 
 import glob
 from pathlib import Path
-from typing import Any, Dict
+from typing import Any, Dict, List, Literal, Mapping, Optional
 
+import pydantic
 import yaml
 from craft_cli import emit
 
+from rockcraft.errors import ProjectValidationError
+
+
+class HttpCheck(pydantic.BaseModel):
+    """Lightweight schema validation for a Pebble HTTP check."""
+
+    url: str
```

Comment:
```suggestion
    url: pydantic.AnyHttpUrl
```

Not sure if you want to be [that specific](https://docs.pydantic.dev/1.10/usage/types/#urls) though.